src/shared/utils/hooks/useSessionBanners/useSessionBanners.ts

```diff
@@ -9,7 +9,7 @@ export const useSessionBanners = () => {
 
   const prevIsAuthorized = useRef(isAuthorized);
 
-  useEffect(() => {
+  useMemo(() => {
```

Comment:
I can't say I actually understand what this change is doing, even after reading your PR description. I get that with a `useMemo`, the callback is executed inline immediately the first time it's encountered by the interpreter, whereas with `useEffect`, the callback is called async after the component has completed the initial render. Could you walk me through exactly what was going wrong here when `useEffect` was in place?

I've also never used `useMemo` for a situation where the return value was discarded, but maybe I just haven't stumbled upon this use-case before now.

The [React docs](https://react.dev/reference/react/useMemo#every-time-my-component-renders-the-calculation-in-usememo-re-runs) on `useMemo` say:
> **You should only rely on `useMemo` as a performance optimization.** If your code doesn’t work without it, find the underlying problem and fix it first. Then you may add useMemo to improve performance.

I guess I'm also wondering if this code aligns with that rule or not…

Reply:
@farmerpaul that's a good callout. With further debugging, I think there may be a re-rendering issue stemming from the redux logic w/ `user`, `token`, and `isAuthorized` state. I can't quite pin down what's triggering the re-renders. 🤔

Reply:
Removing the `useEffect` does ensure the banner isn't automatically cleared because of that particular component re-rendering. I do still have some concerns about the amount of re-rendering taking place, which seems to be a performance issue already established on prod.

I think the rerendering we're seeing is coming from how far along the app renders down the tree as the already authenticated user before it determines that the new user isn't authenticated.